packages/cactus-cmd-api-server/src/test/typescript/benchmark/artillery-api-benchmark.test.ts

```diff
@@ -0,0 +1,143 @@
+} from "../../../main/typescript/public-api";
+
+import { DefaultApi as ApiServerApi } from "../../../main/typescript/public-api";
+
+const log = LoggerProvider.getOrCreate({
+  level: "INFO",
+  label: "logger-test",
+});
+
+const util = require("util");
+const fs = require("fs");
+const exec = util.promisify(require("child_process").exec);
+
+const artilleryScriptLocation =
+  "./packages/cactus-cmd-api-server/src/test/resources/artillery/benchmark-script.yaml";
+const logLevel: LogLevelDesc = "INFO";
+
+test("Start API server, and run Artillery benchmark test.", async (t: Test) => {
+  const keyPair = await JWK.generate("EC", "secp256k1", { use: "sig" }, true);
+  const keyPairPem = keyPair.toPEM(true);
+  const db: ConsortiumDatabase = {
+    cactusNode: [],
+    consortium: [],
+    consortiumMember: [],
+    ledger: [],
+    pluginInstance: [],
+  };
+
+  log.info("Generating Config...");
+
+  const configService = new ConfigService();
+  const apiServerOptions = configService.newExampleConfig();
+  apiServerOptions.authorizationProtocol = AuthorizationProtocol.NONE;
+  apiServerOptions.configFile = "";
+  apiServerOptions.apiCorsDomainCsv = "*";
+  apiServerOptions.apiPort = 4000;
+  apiServerOptions.cockpitPort = 0;
+  apiServerOptions.apiTlsEnabled = false;
+  apiServerOptions.logLevel = "info";
+  apiServerOptions.plugins = [
+    {
+      packageName: "@hyperledger/cactus-plugin-keychain-memory",
+      type: PluginImportType.LOCAL,
+      options: {
+        instanceId: uuidv4(),
+        keychainId: uuidv4(),
+        logLevel,
+      },
+    },
+    {
+      packageName: "@hyperledger/cactus-plugin-consortium-manual",
+      type: PluginImportType.LOCAL,
+      options: {
+        instanceId: uuidv4(),
+        keyPairPem: keyPairPem,
+        consortiumDatabase: db,
+      },
+    },
+  ];
+
+  const config = configService.newExampleConfigConvict(apiServerOptions);
+
+  log.info("Creating API Server...");
+  const apiServer = new ApiServer({
+    config: config.getProperties(),
+  });
+
+  log.info("Starting API Server...");
+
+  const startResponse = apiServer.start();
+  await t.doesNotReject(
+    startResponse,
+    "failed to start API server with dynamic plugin imports configured for it...",
+  );
+  t.ok(startResponse, "startResponse truthy OK");
+
+  const addressInfoApi = (await startResponse).addressInfoApi;
+  const protocol = apiServerOptions.apiTlsEnabled ? "https" : "http";
+  const { address, port } = addressInfoApi;
+  const apiHost = `${protocol}://${address}:${port}`;
+
+  const apiClient = new ApiServerApi({ basePath: apiHost });
+
+  const res = await apiClient.getPrometheusExporterMetricsV1();
+
+  t.equal(res.status, 200);
+
+  // Sets apihost in process env variables to artillery script can read it.
+  process.env.apiHost = apiHost;
+  await fireArtilleryCommand(t);
+
+  // Reading in Artillery JSON output
+  let report = null;
+  try {
+    const jsonString = fs.readFileSync("./report.json");
```

Comment:
@travis-payne Please also dump the whole JSON report to the logs just in case we want to take a deeper look into the results/debug something/etc.